pyccl/pk2d.py

```diff
@@ -182,21 +229,16 @@ def from_model(cls, cosmo, model):
 
     @classmethod
     @functools.wraps(from_model)
+    @deprecated(new_function=from_model.__func__)
     def pk_from_model(cls, cosmo, model):
         return cls.from_model(cosmo, model)
 
     @_Pk2D_descriptor
-    def apply_halofit(self, cosmo, pk_linear=None):
+    @warn_api
+    def apply_halofit(self, cosmo, *, pk_linear=None):
         """Pk2D constructor that applies the "HALOFIT" transformation of
-        Takahashi et al. 2012 (arXiv:1208.2701) on an input linear
-        power spectrum in `pk_linear`.
-
-        Args:
-            cosmo (:class:`~pyccl.core.Cosmology`):
-                A Cosmology object.
-            pk_linear (:class:`Pk2D`, optional):
-                A :class:`Pk2D` object containing the linear power spectrum
-                to transform.
+        Takahashi et al. 2012 (arXiv:1208.2701) on an input linear power
+        spectrum in `pk_linear`. See ``Pk2D.apply_nonlin_model`` for details.
```

Comment:
Well, I don't think this function exists, either you forgot to copy it, or you decided it wasn't worth it. I don't mind either way.